mdsuite/transformations/integrated_heat_current.py

```diff
@@ -24,125 +24,39 @@
-            ke_path = str.encode(join_path(species, "KE"))
-            pe_path = str.encode(join_path(species, "PE"))
-            system_current += tf.reduce_sum(
-                data[positions_path] * (data[ke_path] + data[pe_path]), axis=0
-            )
-
-        return tf.convert_to_tensor(system_current)
-
-    def _prepare_database_entry(self) -> dict:
-        """
-        Add the relevant tensor_values sets and groups in the database_path
-
-        Returns
-        -------
-        data_structure : dict
-                tensor_values structure for use in saving the tensor_values to
-                the database_path.
-        """
-
-        number_of_configurations = self.experiment.number_of_configurations
-        path = join_path(
-            "Integrated_Heat_Current", "Integrated_Heat_Current"
-        )  # name of the new database_path
-        dataset_structure = {path: (number_of_configurations, 3)}
-        self.database.add_dataset(
-            dataset_structure
-        )  # add a new dataset to the database_path
-        data_structure = {path: {"indices": np.s_[:], "columns": [0, 1, 2]}}
-
-        return data_structure
-
-    def _compute_thermal_conductivity(self):
-        """
-        Loop over batches and compute the dipole moment
-        Returns
-        -------
-        Performs the analysis and updates the database.
-        """
-        data_structure = self._prepare_database_entry()
-        type_spec = {}
-
-        species_path = [
-            join_path(species, "Unwrapped_Positions")
-            for species in self.experiment.species
-        ]
-        ke_path = [join_path(species, "KE") for species in self.experiment.species]
-        pe_path = [join_path(species, "PE") for species in self.experiment.species]
-        data_path = list(np.concatenate((species_path, ke_path, pe_path)))
-
-        self._prepare_monitors(data_path)
-        type_spec = self._update_species_type_dict(type_spec, species_path, 3)
-        type_spec = self._update_species_type_dict(type_spec, ke_path, 1)
-        type_spec = self._update_species_type_dict(type_spec, pe_path, 1)
-        type_spec[str.encode("data_size")] = tf.TensorSpec(None, dtype=tf.int16)
-
-        batch_generator, batch_generator_args = self.data_manager.batch_generator(
-            dictionary=True, remainder=True
-        )
-        data_set = tf.data.Dataset.from_generator(
-            batch_generator, args=batch_generator_args, output_signature=type_spec
+        super(IntegratedHeatCurrent, self).__init__(
+            input_properties=[
+                mdsuite_properties.unwrapped_positions,
+                mdsuite_properties.kinetic_energy,
+                mdsuite_properties.potential_energy,
+            ],
+            output_property=mdsuite_properties.integrated_heat_current,
+            batchable_axes=[1, 2],
+            scale_function={"linear": {"scale_factor": 2}},
```

Comment:
scale functions could also be a dataclass

Reply:
it could also be just a function (or lambda) then we do not need any intermediate agents to calculate memory consumption

Reply:
A lambda function would be nice and quite flexible. @SamTov iirc you added that, do you think something like `scale_function=lambda x: 2*x` would also work?

Reply:
Yeah I think I prefer the data class for a reduction in code re-implementation. For all calculators, transformations, and anything loading data we have some scale function. Of course, it isn't a lot of additional code but if we have a data-class or these dicts then you don't have to re-write the function each time. 

If we think that having a lambda on each one is better than having them all sitting in one place then we can do that, but it is in any case for another PR.